### PR TITLE
Add unit tests for kv/importer and restore/checkpoints, plus some bug fixes

### PR DESCRIPTION
**Pull request by kennytm**
### What problem does this PR solve? 

Improve unit test coverage of `restore` package, part 1 of N.

### What is changed and how it works?

Add unit tests (2% → 29%).

Also fixed some minor bugs discovered while creating the unit tests.

### Check List 

Tests 

 - Unit test

Side effects

Related changes

### CODE REVIEW

**Comment by kennytm** [comment]
PTAL @lonng @july2993

File: lightning/kv/importer.go [hunk 1]
```diff
@@ -34,8 +34,8 @@ import (
 )
 
 const (
-	maxRetryTimes    int = 3 // tikv-importer has done retry internally. so we don't retry many times.
-	retryBackoffTime     = time.Second * 3
+	maxRetryTimes           int = 3 // tikv-importer has done retry internally. so we don't retry many times.
```
Comment (lonng) [inline lightning/kv/importer.go]: The type annotation `int` could be removed. And keep aligning with L38 will much pretty.

File: lightning/kv/importer.go [hunk 2]
```diff
@@ -85,15 +86,18 @@ func NewImporter(ctx context.Context, importServerAddr string, pdAddr string) (*
...
+		pdAddr:           pdAddr,
+		retryBackoffTime: defaultRetryBackoffTime,
 	}, nil
 }
 
 // Close the importer connection.
 func (importer *Importer) Close() {
-	importer.conn.Close()
+	if importer.conn != nil {
+		importer.conn.Close()
```
Comment (lonng) [inline lightning/kv/importer.go]: Should we return the error to the caller?

Comment (kennytm) [reply lightning/kv/importer.go]: Added a log on error. Not going to return the error to caller since no caller cares anyway.

Comment (lonng) [reply lightning/kv/importer.go]: OK

**Comment by kennytm** [comment]
PTAL @july2993

**Comment by kennytm** [comment]
Ping @july2993

File: lightning/restore/tidb_test.go [hunk 3]
```diff
@@ -107,3 +147,188 @@ func (s *tidbSuite) TestCreateTableIfNotExistsStmt(c *C) {
...
+	ctx := context.Background()
+
+	s.mockDB.
+		ExpectExec("CREATE DATABASE IF NOT EXISTS `db`").
+		WillReturnResult(sqlmock.NewResult(1, 1))
+	s.mockDB.
+		ExpectExec("USE `db`").
+		WillReturnResult(sqlmock.NewResult(0, 0))
+	s.mockDB.
+		ExpectExec("\\QCREATE TABLE IF NOT EXISTS `t1` (`a` INT PRIMARY KEY,`b` VARCHAR(200));\\E").
```
Comment (july2993) [inline lightning/restore/tidb_test.go]: can you give reference about usage of `\\Q` and `\\E` for me to learn it

Comment (kennytm) [reply lightning/restore/tidb_test.go]: @july2993 https://godoc.org/regexp/syntax Under "Escape sequences"

```
\Q...\E        literal text ... even if ... has punctuation
```

Basically `"\\Qstuff\\E"` is equivalent to `regexp.QuoteMeta("stuff")`.

File: lightning/kv/importer_test.go [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+	mockClient *MockImportKVClient
+	mockWriter *MockImportKV_WriteEngineClient
+	importer   *Importer
+}
+
+var _ = Suite(&importerSuite{})
+
+const testPDAddr = "pd-addr:2379"
+
+// FIXME: Cannot use the real SetUpTest/TearDownTest to set up the mock
```
Comment (july2993) [inline lightning/kv/importer_test.go]: can't catch the point

**Pull request merged**